torch/_inductor/ir.py

```diff
@@ -6528,6 +6528,71 @@ def _has_aliased_buffers(buffers: Sequence[IRNode]) -> bool:
     return len(OrderedSet(id(buffer) for buffer in buffers)) < len(buffers)
 
 
+@ir_dataclass
+class InvokeSubgraph(ExternKernel):
```

Comment:
Do we have anything in place that will require similar aliasing relationships ? would you mind adding a test that shows we dont try to say inplace an input in the subgraph ? 

Or similarly, invoke a subgraph once with the same buffer appearing twice, and then with the subgraph invoked and two different buffers, and check that we dont dedupe the tensor arg internally.

Reply:
https://github.com/pytorch/pytorch/pull/138036 is for aliasing and input mutation.

Dynamo will ensure to not even create an `invoke_subgraph` HOP in fx graph if there is mutation.